File: level2/큰_수_만들기/solution.js
function solution(number, k) {
  let stack = [];
  for (let i = 0; i < number.length; i++) {
    const n = number[i];
    while (k > 0 && n > stack[stack.length - 1]) {
      stack.pop();
      k--;
    }
    stack.push(n);
  }
  let answer = stack.join("");
  return answer;
}

console.log(solution("1924", 2));
console.log(solution("1231234", 3));
console.log(solution("4177252841", 4));

File: level2/큰_수_만들기/solution.rb
def solution(number, k)
  stack = []

  number.chars.each do |n|
    while !stack.empty? && stack.last < n && k > 0
      stack.pop
      k -= 1
    end
    stack << n
  end

  answer = stack[0...(number.size - k)].join
  return answer
end

p solution "1924", 2
p solution "1231234", 3
p solution "4177252841", 4
p solution "9999999999", 4
